source/contributing/how-to-get-help.md

```diff
@@ -3,40 +3,54 @@
   attribute.
-- Check out if any teams are responsible for the relevant subsystem:
+- Check if any teams are responsible for the relevant subsystem:
   - On the [NixOS website](https://nixos.org/community/#governance-teams).
   - In the [list of Nixpkgs maintainer teams](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix).
   - In the `CODEOWNERS` files for [Nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS) or
     [Nix](https://github.com/NixOS/nix/blob/master/.github/CODEOWNERS).
-- Look at the [git log](https://www.git-scm.com/docs/git-log) for the files you need help
-  with, and take note of the email addresses of people who committed.
+- Check the output of [`git blame`](https://git-scm.com/docs/git-blame) or [`git log`](https://www.git-scm.com/docs/git-log) for the files you need help with.
+  Take note of the email addresses of people who committed relevant code.
+
+## Which communication channels to use
 
-Once you've found the people you're looking for, you can contact them via:
+Once you've found the people you're looking for, you can contact them on one of the following platforms:
 
 - GitHub
 
-  [Mention the GitHub username](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#mentioning-people-and-teams) found in the [`maintainers-list.nix` file][maintainers-list] in issue or pull request descriptions, or comments.
+  All the source code is maintained on GitHub.
+  This is the right place to discuss implementation details.
+
+  In issue comments or pull request descriptions, [mention the GitHub username](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#mentioning-people-and-teams) found in the [`maintainers-list.nix` file][maintainers-list].
 
 - [Discourse](https://discourse.nixos.org)
 
+  Discourse is used for announcements, coordination, and open-ended questions.
+
   Try the GitHub username found in the [`maintainers-list.nix` file][maintainers-list] to mention or directly contact a specific user.
   Note that some people use a different username on Discourse.
 
 - [Matrix]
 
-  Use the Matrix handle found in the [`maintainers-list.nix` file][maintainers-list].
-  If no Matrix handle is present for a specific maintainer, try searching for them using their GitHub username, as most people tend to use the same one across channels.
+  Matrix is used for direct messages and short-lived, timely exchanges.
+
+  To contact a maintainer, use their Matrix handle found in the [`maintainers-list.nix` file][maintainers-list].
+  If no Matrix handle is present for a specific maintainer, try searching for their GitHub username, as most people tend to use the same one across channels.
+
+  Maintainer teams usually have their own public Matrix room.
```

Comment:
I think Matrix rooms should generally be preferred over Matrix DMs, so I'd mention those before. Also I'd weaken this to just:
```suggestion
  Maintainer teams sometimes have their own public Matrix room.
```